components/src/components/molecules/PageButtons/PageButtons.tsx

```diff
@@ -77,38 +93,47 @@ export const PageButtons = ({
     Math.min(Math.max(current - maxPerSide, 1), total - max + 1),
     1,
   )
-  const array = Array.from({ length: max }, (_, i) => start + i).filter(
+  const pageNumbers = Array.from({ length: max }, (_, i) => start + i).filter(
     (x) => x <= total,
   )
 
   if (total > max) {
     if (alwaysShowFirst && start > 1) {
-      array[0] = -1
-      array.unshift(1)
-    } else if (start > 1) {
-      array.unshift(-1)
+      if (showElipsis) {
+        pageNumbers[0] = Elipsis.left
+        pageNumbers.unshift(1)
+      } else {
+        pageNumbers[0] = 1
+      }
+    } else if (showElipsis && start > 1) {
+      pageNumbers.unshift(Elipsis.left)
     }
 
     if (alwaysShowLast && total > current + maxPerSide) {
-      array[array.length - 1] = -1 * total
-      array.push(total)
-    } else if (total > current + maxPerSide) {
-      array.push(-1 * total)
+      if (showElipsis) {
+        pageNumbers[pageNumbers.length - 1] = Elipsis.right
+        pageNumbers.push(total)
+      } else {
+        pageNumbers[pageNumbers.length - 1] = total
+      }
+    } else if (showElipsis && total > current + maxPerSide) {
+      pageNumbers.push(Elipsis.right)
     }
   }
 
   return (
     <Container
       {...{ ...props, 'data-testid': getTestId(props, 'pagebuttons') }}
     >
-      {array.map((value) =>
-        0 > value ? (
+      {pageNumbers.map((value) =>
+        value < 0 ? (
```

Comment:
```suggestion
        value === Ellipsis.left || value === Ellipsis.right ? (
```
I think this still clearer